eth/downloader/peer.go

```diff
@@ -131,6 +167,36 @@ func (p *peer) SetIdle() {
 	atomic.StoreInt32(&p.idle, 0)
 }
 
+// SetIdle sets the peer to idle, allowing it to execute new retrieval requests.
```

Comment:
Is this throttling of bandwidth really needed? Does this not to some extent re-invent flow control at lower layers like TCP? What are the implications if we do not have this throttling? Is there a risk that a peer demoted due to bad bandwidth cannot regain good allowance if other peers where never demoted?